worker/raft/worker.go

```diff
@@ -170,6 +194,13 @@ func NewWorker(config Config) (*Worker, error) {
 	}); err != nil {
 		return nil, errors.Trace(err)
 	}
+	// Wait for the loop to be started.
+	select {
+	case <-config.Clock.After(LoopTimeout):
+		w.catacomb.Kill(ErrStartTimeout)
+		return nil, ErrStartTimeout
+	case <-w.raftCh:
```

Comment:
Where is the event sent for this call?

Reply:
From `loop()` - the problem was that the call to `raft.NewRaft` was hanging (because of a similar problem in the stream layer), so the loop method was never actually getting into the select loop where it would do the send.